tests/functional/olp-cpp-sdk-dataservice-read/DataserviceReadVersionedLayerClientTestMockServer.cpp

```diff
@@ -0,0 +1,133 @@
+        CreateDefaultNetworkRequestHandler();
+
+    auto appid =
+        CustomParameters::getArgument("dataservice_read_test_versioned_appid");
+    auto secret =
+        CustomParameters::getArgument("dataservice_read_test_versioned_secret");
+    olp::authentication::Settings auth_settings({appid, secret});
+    auth_settings.network_request_handler = network;
+
+    olp::authentication::TokenProviderDefault provider(auth_settings);
+    olp::client::AuthenticationSettings auth_client_settings;
+    auth_client_settings.provider = provider;
+
+    settings_ = std::make_shared<olp::client::OlpClientSettings>();
+    settings_->network_request_handler = network;
+
+    settings_->authentication_settings = auth_client_settings;
+    SetUpMockServer(network);
+  }
+
+  void TearDown() override {
+    auto network = std::move(settings_->network_request_handler);
+    settings_.reset();
+    mock_server_client_.reset();
+    // when test ends we must be sure that network pointer is not captured
+    // anywhere
+    ASSERT_EQ(network.use_count(), 1);
+  }
+
+  void SetUpMockServer(std::shared_ptr<olp::http::Network> network) {
+    // setup proxy
+    settings_->proxy_settings =
+        olp::http::NetworkProxySettings()
+            .WithHostname(kMockServerHost)
+            .WithPort(kMockServerPort)
+            .WithType(olp::http::NetworkProxySettings::Type::HTTP);
+
+    // create client to set mock server expectations
+    olp::client::OlpClientSettings olp_client_settings;
+    olp_client_settings.network_request_handler = network;
+    mock_server_client_ = std::make_shared<mockserver::MockServerHelper>(
+        olp_client_settings, GetTestCatalog());
+  }
+
+  std::string GetTestCatalog() {
+    return CustomParameters::getArgument("dataservice_read_test_catalog");
+  }
+
+  std::shared_ptr<olp::client::OlpClientSettings> settings_;
+  std::shared_ptr<mockserver::MockServerHelper> mock_server_client_;
+};
+
+TEST_F(DataserviceReadVersionedLayerClientTestMockServer, GetPartitions) {
+  olp::client::HRN hrn(GetTestCatalog());
+  {
+    mock_server_client_->MockTimestamp();
+    mock_server_client_->MockAuth();
+
+    mock_server_client_->MockGetResponce(
+        mockserver::DefaultResponces::GenerateApis(GetTestCatalog()));
+
+    mock_server_client_->MockGetResponce(
+        mockserver::DefaultResponces::GenerateVersion(44));
+
+    mock_server_client_->MockGetResponce(
+        mockserver::DefaultResponces::GeneratePartitions(4));
+  }
+
+  auto catalog_client =
+      std::make_unique<olp::dataservice::read::VersionedLayerClient>(
+          hrn, "testlayer", boost::none, *settings_);
+
+  auto request = olp::dataservice::read::PartitionsRequest();
+  auto future = catalog_client->GetPartitions(request);
+  auto partitions_response = future.GetFuture().get();
+
+  EXPECT_SUCCESS(partitions_response);
+  ASSERT_EQ(4u, partitions_response.GetResult().GetPartitions().size());
```

Comment:
Please also check the expectations on the server. Make sure only the requests you setup were triggered and only one call for each.